adapters/handlers/grpc/v1/service.go

```diff
@@ -49,13 +50,39 @@ type Service struct {
 	config               *config.Config
 	authorizer           authorization.Authorizer
 	logger               logrus.FieldLogger
+
+	authenticator      *authHandler
+	batchHandler       *batch.Handler
+	batchQueuesHandler *batch.QueuesHandler
 }
 
 func NewService(traverser *traverser.Traverser, authComposer composer.TokenFunc,
 	allowAnonymousAccess bool, schemaManager *schemaManager.Manager,
 	batchManager *objects.BatchManager, config *config.Config, authorization authorization.Authorizer,
-	logger logrus.FieldLogger,
+	logger logrus.FieldLogger, shutdown *batch.Shutdown,
 ) *Service {
+	authenticator := NewAuthHandler(allowAnonymousAccess, authComposer)
+	internalQueue := batch.NewBatchInternalQueue()
+	batchWriteQueues := batch.NewBatchWriteQueues()
+	batchReadQueues := batch.NewBatchReadQueues()
+
+	batchHandler := batch.NewHandler(authorization, batchManager, logger, authenticator, schemaManager)
+	batchQueuesHandler := batch.NewQueuesHandler(shutdown.HandlersCtx, shutdown.SendWg, shutdown.StreamWg, shutdown.ShutdownFinished, batchWriteQueues, batchReadQueues, logger)
+
+	var numWorkers int
+	numWorkersStr := os.Getenv("GRPC_BATCH_WORKERS_COUNT")
+	if numWorkersStr != "" {
+		x, err := strconv.Atoi(numWorkersStr)
+		if err == nil {
+			numWorkers = x
+		}
+	} else {
+		numWorkers = 4
+	}
```

Comment:
I would remove this setting and rely solely on `var _NUMCPU = runtime.GOMAXPROCS(0)` wdyt?

Reply:
and maybe if we would see it beneficial to have this setting configurable I would then introduce it

Reply:
Sure! My concern was that it could potentially overwhelm the server if there are many workers running but this isn't tested at all. We can reintroduce a specific env var if that ends up being the case during testing